autotest/ogr/ogr_gpkg.py

```diff
@@ -3884,6 +3884,80 @@ def test_ogr_gpkg_json():
 
     gdal.Unlink(filename)
 
+
+###############################################################################
+# Test unique constraints on fields
+
+
+def test_ogr_gpkg_unique():
+
+    gdaltest.gpkg_dr = ogr.GetDriverByName('GPKG')
+    if gdaltest.gpkg_dr is None:
+        pytest.skip()
+
+    if gdaltest.is_travis_branch('trusty_32bit'):
+        pytest.skip('gcc too old')
+
+    ds = gdaltest.gpkg_dr.CreateDataSource('/vsimem/ogr_gpkg_unique.gpkg')
+    lyr = ds.CreateLayer('test', geom_type=ogr.wkbNone)
+
+    # Default: no unique constraints
+    field_defn = ogr.FieldDefn('field_default', ogr.OFTString)
+    lyr.CreateField(field_defn)
+
+    # Explicit: no unique constraints
+    field_defn = ogr.FieldDefn('field_no_unique', ogr.OFTString)
+    field_defn.SetUnique(0)
+    lyr.CreateField(field_defn)
+
+    # Explicit: unique constraints
+    field_defn = ogr.FieldDefn('field_unique', ogr.OFTString)
+    field_defn.SetUnique(1)
+    lyr.CreateField(field_defn)
+
+    # Now check for getters
+    layerDefinition = lyr.GetLayerDefn()
+    fldDef = layerDefinition.GetFieldDefn(0)
+    assert not fldDef.IsUnique()
+    fldDef = layerDefinition.GetFieldDefn(1)
+    assert not fldDef.IsUnique()
+    fldDef = layerDefinition.GetFieldDefn(2)
+    assert fldDef.IsUnique()
+
+    ds = None
+
+    gdal.Unlink('/vsimem/ogr_gpkg_unique.gpkg')
+
+    # Now test loading the same layer from disk, constraints should be
+    # automatically set on field 2
+    ds = ogr.Open('data/ogr_gpkg_unique.gpkg')
```

Comment:
I actually need to recreate the whole table from SQL because the create instruction explicitly test for different quoting styles and spacings in fields and index names.

I'm trying but I cannot get it done:

```python
    # Now create another layer from SQL to test quoting of fields
    # and indexes
    # Note: leave create table in a single line because of regex spaces testing
    sql = """
    CREATE TABLE IF NOT EXISTS "test2" ( "fid" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "field_default" TEXT, "field_no_unique" TEXT, "field_unique" TEXT UNIQUE,`field unique2` TEXT UNIQUE,field_unique3 TEXT UNIQUE, "field_unique_index" TEXT, `field unique index2`, "field_unique_index3" TEXT);
    CREATE UNIQUE INDEX test2_unique_idx ON test2(field_unique_index);
    CREATE UNIQUE INDEX test2_unique_idx2 ON test2(`field unique index2`);
    CREATE UNIQUE INDEX test2_unique_idx3 ON test2("field_unique_index3");
    INSERT INTO gpkg_contents VALUES('test2','attributes','test2','','2020-05-27T12:27:30.136Z',NULL,NULL,NULL,NULL,0);
    INSERT INTO gpkg_ogr_contents VALUES('test2',NULL);
    """

    ds.ExecuteSQL(sql)

    # Now test loading the same layer from disk, check layer test2
    ds = ogr.Open('/tmp/ogr_gpkg_unique.gpkg')

    assert ds.GetLayerCount() == 2   # <----- fails here!
```

Reply:
or put the various lines in a python array and iterate over it